### PR TITLE
[3.0] Various deprecation removal

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by sstok** [comment]
LGTM :-1:

File: src/Symfony/Component/Config/Definition/Builder/BooleanNodeDefinition.php [hunk 1]
```diff
@@ -31,18 +31,6 @@ public function __construct($name, NodeParentInterface $parent = null)
     }
 
     /**
-     * {@inheritdoc}
-     *
-     * @deprecated Deprecated since version 2.8, to be removed in 3.0.
-     */
-    public function cannotBeEmpty()
```
Comment (Koc) [inline src/Symfony/Component/Config/Definition/Builder/BooleanNodeDefinition.php]: As I understand correctly - after removing this overridion, calling this method just executes from parent class without any error?

Comment (WouterJ) [inline src/Symfony/Component/Config/Definition/Builder/BooleanNodeDefinition.php]: @Koc, there is no emptiness in a boolean and numberic value. See https://github.com/symfony/symfony/pull/15285 and https://github.com/symfony/symfony/pull/15170

File: src/Symfony/Bridge/Doctrine/Tests/Form/Type/EntityTypePerformanceTest.php [hunk 2]
```diff
@@ -90,7 +90,7 @@ public function testCollapsedEntityField()
         $this->setMaxRunningTime(1);
 
         for ($i = 0; $i < 40; ++$i) {
-            $form = $this->factory->create('entity', null, array(
+            $form = $this->factory->create('Symfony\Bridge\Doctrine\Form\Type\EntityType', null, array(
```
Comment (Tobion) [inline src/Symfony/Bridge/Doctrine/Tests/Form/Type/EntityTypePerformanceTest.php]: this should be done in 2.8

Comment (nicolas-grekas) [inline src/Symfony/Bridge/Doctrine/Tests/Form/Type/EntityTypePerformanceTest.php]: It is on 2.7 on #15687

**Comment by Tobion** [comment]
:+1: 
Status: Reviewd

**Comment by Tobion** [comment]
Status: Reviewed

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/Config/Definition/Builder/BooleanNodeDefinition.php (hunk 1 above)
Comment (Ener-Getick) [inline src/Symfony/Component/Config/Definition/Builder/BooleanNodeDefinition.php]: Wasn't this supposed to throw an exception ?  (see [this message](https://github.com/symfony/symfony/pull/15170#issuecomment-119538615))

Comment (nicolas-grekas) [inline src/Symfony/Component/Config/Definition/Builder/BooleanNodeDefinition.php]: @Ener-Getick please open an issue to get an answer

Comment (Ener-Getick) [inline src/Symfony/Component/Config/Definition/Builder/BooleanNodeDefinition.php]: See #16812

**Pull request merged**